core/commonMain/src/ArgParser.kt

```diff
@@ -213,23 +220,23 @@ open class ArgParser(
     private fun inspectRequiredAndDefaultUsage() {
         var previousArgument: ParsingValue<*, *>? = null
         arguments.forEach { (_, currentArgument) ->
-            // TODO: 'let' body never executes
-            previousArgument?.let {
+            previousArgument?.let { previous ->
                 // Previous argument has default value.
-                it.descriptor.defaultValue?.let {
-                    if (currentArgument.descriptor.defaultValue == null && currentArgument.descriptor.required) {
-                        printError("Default value of argument ${previousArgument.descriptor.fullName} will be unused,  " +
+                if (previous.hasDefaultValue) {
+                    if (!currentArgument.hasDefaultValue && currentArgument.descriptor.required) {
+                        printError("Default value of argument ${previous.descriptor.fullName} will be unused,  " +
```

Comment:
`printError` should not be used for reacting on parser configuration errors. Printing an error together with the usage info only makes sense for a program **user**'s error, rather than for a program **author**'s error.

Please review all usages of `printError`.